### PR TITLE
Add new output module for Splunk

### PR DESCRIPTION
**Pull request by w0Tx**
This PR include a new way to export data to a splunk instance through HTTP Event Collector.

I've decided to create a new output module instead of modifying the HTTP output module to keep the code as simple as possible.

I've also added the documentation associated with it.

### CODE REVIEW

**Comment by TheTechromancer** [comment]
Thanks for doing this! The module looks solid, we just need to write a test for it.

**Review by TheTechromancer — APPROVED** [review]

**Comment by TheTechromancer** [comment]
Tests written and passing, waiting for approval.

**Review by liquidsec — APPROVED** [review]

**Pull request merged**
